src/hooks/useNavigatorShare.ts

```diff
@@ -0,0 +1,55 @@
+interface IShareData {
+  title: string
+  text: string
+  url?: string
+  files?: File[]
+}
+
+const errorMessages: Record<string, string> = {
+  NotAllowedError: 'Permiso para compartir denegado.',
+  AbortError: 'La acción de compartir fue abortada.',
+  NotSupportedError: 'Tu navegador no soporta la función de compartir.',
+  TypeError: 'Error al compartir: tipo de dato incorrecto.'
+}
+
+function checkPermission (files?: File[]) {
+  if (!navigator.canShare) {
+    throw new Error('Tu navegador no soporta la función de compartir.')
+  }
+
+  if (!navigator.canShare({ files } ?? { files: [new File([], '')] })) {

```

Comment:
Instead of the `??`, I think here goes `||` since it is inside of an if statement.